lib/config/executor/network_interface.go

```diff
@@ -60,17 +60,22 @@ type ContainerNetwork struct {
 
 	// The network scope the IP belongs to.
 	// The IP address is the default gateway
-	Gateway net.IPNet `vic:"0.1" scope:"read-write" key:"gateway"`
+	Gateway net.IPNet `vic:"0.1" scope:"read-only" key:"gateway"`
 
 	// Should this gateway be the default route for containers on the network
 	Default bool `vic:"0.1" scope:"read-only" key:"default"`
 
 	// The set of nameservers associated with this network - may be empty
-	Nameservers []net.IP `vic:"0.1" scope:"read-write" key:"dns"`
+	Nameservers []net.IP `vic:"0.1" scope:"read-only" key:"dns"`
 
 	// The IP ranges for this network
 	Pools []ip.Range `vic:"0.1" scope:"read-only" key:"pools"`
 
 	// set of network wide links and aliases for this container on this network
 	Aliases []string `vic:"0.1" scope:"hidden" key:"aliases"`
+
+	Assigned struct {
+		Gateway     net.IPNet `vic:"0.1" scope:"read-write" key:"gateway"`
```

Comment:
for what it's worth, you don't need to have the annotations on the elements - they'll inherit from the struct